src/main/java/dk/bec/polonez/reservationsystem/service/DiscountService.java

```diff
@@ -16,4 +23,30 @@ public DiscountService(DiscountRepository discountRepository) {
     public int getByCode(String code){
         return discountRepository.getByCode(code);
     }
+
+    public Discount findById(int id){
+        Optional<Discount> optionalDiscount = discountRepository.findById(id);
+        return optionalDiscount.
+                orElseThrow(()->new ResponseStatusException(HttpStatus.NOT_FOUND));
```

Comment:
raczej z serwisu bym nie rzucal wyjatku ResponseStatusException <-- to powinna byc odpowiedzialnosc controllera

Reply:
I agree, I added similar comment in previous pull-request #5 regarding ```AuthService```.